<?xml version="1.0"?>
<!-- 
  Wrapper launch file for the lebai robot_state node.
-->
<launch>
  <!-- IP of robot (or PC running simulation) -->
  <arg name="robot_ip" doc="IP of controller" />
  <arg name="robot_ns" default=""/>
  <arg name="use_ns" default="0"/>

  <param name="robot_ip_address" type="str" value="$(arg robot_ip)" unless="$(arg use_ns)"/>
  <param name="$(arg robot_ns)/robot_ip_address" type="str" value="$(arg robot_ip)" if="$(arg use_ns)"/>
  
  <param name="rcs_port" type="int" value="5181"/>

  <node name="system_service" pkg="lebai_driver" type="system_service"  unless="$(arg use_ns)" output="screen">
  </node>    
  <node name="$(arg robot_ns)_system_service" pkg="lebai_driver" type="system_service"  if="$(arg use_ns)" output="screen">
    <remap from="robot_ip_address" to="$(arg robot_ns)/robot_ip_address"/>
  </node>    
</launch>